### PR TITLE
ss58-prefix second-priority data migration

### PR DESCRIPTION
**Pull request by shamilkhan**

### CODE REVIEW

File: backend/crawlers/ss58PrefixMigration.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        logger.warning(loggerOptions, 'args are not a sting');
+        return args;
+    }
+    // Extract list of accounts
+    const accounts = args.match(/5[a-zA-Z0-9]{47}/gm);
+    let result = args;
+
+    if (Array.isArray(accounts)) {
+        accounts.forEach((account) => {
+            result = result.replace(account, decode(account));
```
Comment (shamilkhan) [inline backend/crawlers/ss58PrefixMigration.js]: Add check for accounts and prevent replacing not account value.

File: backend/crawlers/ss58PrefixMigration.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    } else {
+        return keyring.encodeAddress(
+            keyring.decodeAddress(address),
+            config.ss58Prefix);
+    }
+};
+
+const changeAccountInJSON = (args) => {
+    if (typeof args !== 'string') {
+        logger.warning(loggerOptions, 'args are not a sting');
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: Fix typo in `sting`

File: backend/crawlers/ss58PrefixMigration.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    level: backendConfig.logLevel,
+});
+const loggerOptions = {
+    crawler: crawlerName,
+};
+const config = backendConfig.crawlers.find(
+    ({name}) => name === crawlerName,
+);
+
+const decode = (address) => {
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: You have the same code in migration files. Please, follow DRY

File: backend/crawlers/ss58PrefixMigration.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    const batchSize = config.batchSize || 50000;
+
+    while (startBlock < endBlock) {
+        const sql = `SELECT block_number, event_index, data FROM event WHERE method NOT LIKE 'ExtrinsicSuccess' AND data LIKE '%"5%' AND block_number BETWEEN ${startBlock} AND ${startBlock + batchSize};`;
+        const {rowCount, rows} = await dbQuery(client, sql);
+
+        startBlock += batchSize;
+
+        if (!rowCount) {
+            logger.info(loggerOptions, `Skiped for batch ${startBlock} -> ${endBlock + batchSize}`);
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: Fix typo in `Skiped`

File: backend/crawlers/ss58PrefixMigration.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+                }
+            }
+            logger.info(loggerOptions, `Finished batching for ${startBlock} -> ${endBlock + batchSize}`);
+        }
+    }
+
+    logger.info(loggerOptions, '✅ Finished migration for args property for extrinsic table');
+};
+
+const migrateSignerProperty = async (client) => {
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: This and `migrateDataForEvent` methods look very similar. Can use reuse the code?

File: backend/crawlers/ss58PrefixMigration.js [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+};
+
+const crawler = async () => {
+
+    logger.info(loggerOptions, 'Running ss58 prefix migration crawler...');
+
+    const client = await getClient(loggerOptions);
+
+    // Event table
+    await migrateDataForEvent(client);
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: Let's use `table first` approach for methods naming. This one will be renamed into `migrateDataInEventTable`. If you need to migrate few columns, please, use execute methods inside `table` method.

File: backend/crawlers/ss58PrefixMigration.js [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+        await dbQuery(client, `UPDATE block SET block_author='${nextAddress}' WHERE block_author='${block_author}'`, loggerOptions);
+
+        logger.info(loggerOptions, `Finished migration for ${block_author}`);
+    }
+
+    logger.info(loggerOptions, '✅ Finished migration for Block table');
+};
+
+const crawler = async () => {
```
Comment (AndreiNavoichyk) [inline backend/crawlers/ss58PrefixMigration.js]: How long is migration execution? Why we can not execute it as a usual migration?

Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: It runs for ~ 1 hour. Let's move it to migration.

File: backend/crawlers/ss58PrefixMigration.js (hunk 6 above)
Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: Yeap, I used these names: 
```
  await migrateBlockTable(db, ss58Format);
    await migrateExtrinsicTableSigners(db, ss58Format);
    await migrateExtrinsicTableArgs(db, ss58Format);
    await migrateEventTable(db, ss58Format);
```

File: backend/crawlers/ss58PrefixMigration.js (hunk 5 above)
Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: Done.

File: backend/crawlers/ss58PrefixMigration.js (hunk 4 above)
Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: Done.

File: backend/crawlers/ss58PrefixMigration.js (hunk 2 above)
Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: Done.

File: backend/crawlers/ss58PrefixMigration.js (hunk 3 above)
Comment (shamilkhan) [reply backend/crawlers/ss58PrefixMigration.js]: Done. Moved logic to /shared folder

**Review by MRamanenkau — DISMISSED** [review]

File: db/migrations/20230403094334-ss58-prefix-second-priority-data.js [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+};
+
+const migrateBlockTable = async (db, ss58Format) => {
+    console.log('Start migration for Block table');
+
+    const result = await executeDbRunSqlAsPromise(db, `SELECT DISTINCT block_author FROM block WHERE block_author LIKE '5%'`);
+
+    console.log(`Selected ${result.rowCount} accounts from block table`);
+
+    await Promise.all(result.rows.map(async (row) => {
```
Comment (AndreiNavoichyk) [inline db/migrations/20230403094334-ss58-prefix-second-priority-data.js]: Let's do it in series instead of parallelism to avoid spike load on DB.

Comment (shamilkhan) [reply db/migrations/20230403094334-ss58-prefix-second-priority-data.js]: Done.
[Replace Promise.all with sequential await calls in migration code](https://github.com/Cerebellum-Network/cerestats/pull/130/commits/88a09665964b925408c6e5b0c45697f53f34d4bf)

**Review by AndreiNavoichyk — DISMISSED** [review]
Please, update changelog

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**